src/operator/contrib/krprod.h

```diff
@@ -0,0 +1,138 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ *  \file krprod.h
+ *  \brief Core function for Khatri-Rao product
+ *  \author Jencir Lee
+ */
+#ifndef MXNET_OPERATOR_CONTRIB_KRPROD_H_
+#define MXNET_OPERATOR_CONTRIB_KRPROD_H_
+#include <vector>
+#include "mshadow/tensor.h"
+
+namespace mxnet {
+namespace op {
+
+using namespace mshadow;
+using namespace mshadow::expr;
+
+/*!
+ * \brief Computes Khatri-Rao product
+ *
+ * Given the transpose of input matrices, this function computes the
+ * transpose of the Khatri-Rao product of the input matrices.
```

Comment:
It was mainly for the CP Decomposition, as the MXNet matrices are row-major, I found it easiest to store in and perform operations on effectively transposed factor matrices. But you're right the naming is confusing, I'll change it.